crates/viewer/re_space_view_spatial/src/pickable_textured_rect.rs

```diff
@@ -0,0 +1,64 @@
+use re_log_types::EntityPath;
+use re_renderer::{renderer::TexturedRect, QueueableDrawData};
+use re_types::components::DepthMeter;
+use re_viewer_context::{ImageInfo, SpaceViewSystemExecutionError};
+
+pub enum PickableRectSourceData {
+    /// The rectangle is an image with pixel data, potentially some depth meta information.
+    Image {
+        image: ImageInfo,
+        depth_meter: Option<DepthMeter>,
+    },
+
+    /// The rectangle is a frame in a video.
+    Video {
+        // TODO(andreas): add more information here for picking hover etc.
+        resolution: glam::Vec2,
+    },
+
+    /// The rectangle represents an error icon.
+    ErrorPlaceholder {
+        // TODO(andreas): add more information here for picking hover etc.
+        resolution: glam::Vec2,
+    },
+}
+
+/// Image rectangle that can be picked in the view.
+pub struct PickableTexturedRect {
+    /// Path to the image (note image instance ids would refer to pixels!)
+    pub ent_path: EntityPath,
+
+    /// Textured rectangle used by the renderer.
+    pub textured_rect: TexturedRect,
+
+    /// Associated data.
+    pub source_data: PickableRectSourceData,
+}
+
+impl PickableTexturedRect {
+    /// How wide the rectangle is in pixels.
+    pub fn pixel_width(&self) -> u64 {
```

Comment:
We should clarify that this is the resolution of the source data, and not the size of the rectangle on screen.

Also, this could be called `resolution()` and return a 2D `[u64; 2]` in case we ever want the height 🤷